lib/email_api/email_api.py

```diff
@@ -0,0 +1,178 @@
+import os
+import random
+from smtplib import SMTP_SSL
+from email.header import Header
+from email.mime.application import MIMEApplication
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from email.utils import formatdate
+from typing import Dict
+
+
+class EmailApi:
+    def load_template(self, prop, **kwargs):
```

Comment:
Use named params instead of kwargs

Reply:
@DavidFair Because of the way I am selecting the query to run in OpenstackServer, would it be okay to leave the kwargs in there? - they are not used but must be present e.g. search_all_servers doesn't need the day parameter and search_servers_older_than does so the former will cause an error if passed days. The only alternative I can think of is using a massive if statement to switch between the different functions which is a lot more messy.

Reply:
Can we document the supported ones in a doc string then or put the URL to where a future dev can see the supported ones? 

In this case that's fine since we do genuinely have dynamic kwargs

Reply:
Where would be the best place to put this? The supported ones are defined in the enum so it produces a drop down list, otherwise they come from the functions defined in server_action.